File: cflib_2_12/cplus/libcflib/CFLibDependentsDetectedException.cpp
/*
 *  MSS Code Factory CFLib 2.12
 *
 *	Copyright (c) 2020 Mark Stephen Sobkow
 *	
 *	Licensed under the Apache License, Version 2.0 (the "License");
 *	you may not use this file except in compliance with the License.
 *	You may obtain a copy of the License at
 *	
 *	    http://www.apache.org/licenses/LICENSE-2.0
 *	
 *	Unless required by applicable law or agreed to in writing, software
 *	distributed under the License is distributed on an "AS IS" BASIS,
 *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *	See the License for the specific language governing permissions and
 *	limitations under the License.
 */

#include <cflib/CFLib.hpp>
#include <cflib/CFLibDependentsDetectedException.hpp>

namespace cflib {

	const std::string CFLibDependentsDetectedException::S_RELATION( " relation " );
	const std::string CFLibDependentsDetectedException::S_DEPENDENTS_DETECTED_FOR_KEY( " dependents detected for primary key " );

	CFLibDependentsDetectedException::~CFLibDependentsDetectedException()
	{
	}

	CFLibDependentsDetectedException::CFLibDependentsDetectedException(
		const CFLibDependentsDetectedException& src )
	: std::runtime_error( src )
	{
	}

	CFLibDependentsDetectedException::CFLibDependentsDetectedException(
		const std::string& msg )
	: std::runtime_error( msg )
	{ 
	}

	CFLibDependentsDetectedException::CFLibDependentsDetectedException(
		const std::string& throwingClassName,
		const std::string& methName,
		const std::string& msg )
	: std::runtime_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE + msg )
	{
	}

	CFLibDependentsDetectedException::CFLibDependentsDetectedException(
		const std::string& throwingClassName,
		const std::string& methName,
		const std::string& relationType,
		const std::string& relationName,
		ICFLibAnyObj* argKey )
	: std::runtime_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE + S_RELATION + relationName + S_DEPENDENTS_DETECTED_FOR_KEY + (( argKey != NULL ) ? argKey->toString() : CFLib::S_EMPTY ) )
	{
	}
}

File: cflib_2_12/csharp/CFLib/CFLibDependentsDetectedException.cs
/*
 *  MSS Code Factory CFLib 2.11
 *
 *	Copyright (c) 2019 Mark Stephen Sobkow
 *	
 *	Licensed under the Apache License, Version 2.0 (the "License");
 *	you may not use this file except in compliance with the License.
 *	You may obtain a copy of the License at
 *	
 *		http://www.apache.org/licenses/LICENSE-2.0
 *	
 *	Unless required by applicable law or agreed to in writing, software
 *	distributed under the License is distributed on an "AS IS" BASIS,
 *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *	See the License for the specific language governing permissions and
 *	limitations under the License.
 */

using System;
using System.Globalization;

namespace CFLib
{
	public class CFLibDependentsDetectedException : CFLibArgumentException {

		protected String relnType = null;
		protected String relnName = null;
		protected String relnTarget = null;
		protected Object pkey = null;

		public CFLibDependentsDetectedException(
			String msg )
		: base(msg)
		{
		}
		
		public CFLibDependentsDetectedException(
			Type throwingType,
			String methName,
			String msg )
		: base(throwingType, methName, msg)
		{
		}

		public CFLibDependentsDetectedException(
			Type throwingType,
			String methName,
			String msg,
			Exception th )
		: base(throwingType, methName, msg, th)
		{
		}

		public CFLibDependentsDetectedException(
			Type throwingType,
			String methName,
			String relationType,
			String relationName,
			String targetName,
			Object argKey )
		: base(throwingType, methName,
                relationType + " relation " + relationName
					+ " dependents detected"
					+ ((argKey != null ) ? " for primary key " + argKey.ToString() : "" ) )
		{
			relnType = relationType;
			relnName = relationName;
			relnTarget = targetName;
			pkey = argKey;
		}

		public CFLibDependentsDetectedException(
			Type throwingType,
			String methName,
			String relationType,
			String relationName,
			String targetName,
			Object argKey,
			Exception th )
		: base(throwingType, methName,
                relationType + " relation " + relationName
					+ " dependents detected"
					+ ((argKey != null ) ? " for primary key " + argKey.ToString() : "" ),
				th )
		{
			relnType = relationType;
			relnName = relationName;
			relnTarget = targetName;
			pkey = argKey;
		}

		public String getRelationType() {
			return( relnType );
		}
		
		public String getRelationName() {
			return( relnName );
		}
		
		public Object getPrimaryKey() {
			return( pkey );
		}
	}
}
